### ISSUE TITLE
Import/Export of rules for RP 1.xx

### ISSUE THREAD

**Issue by msxfm** [RequestPolicyContinued/requestpolicy#342]
<a href="https://github.com/kronowski"><img src="https://avatars.githubusercontent.com/u/2347635?" align="left" width="96" height="96" hspace="10"></img></a> **Issue by [kronowski](https://github.com/kronowski)**
_Friday Sep 14, 2012 at 15:40 GMT_
_Originally opened as https://github.com/RequestPolicy/requestpolicy/issues/342_

---

Would be nice if  the rules could be exported  and imported like in the pre 1.xx versions of request policy.

**Comment by powerman** [comment]
Is there a way to convert user.json mentioned above into 0.5 format?

**Comment by myrdd** [comment]
@powerman No, RequestPolicy does not have this capability. Of course you could write your own tool, or do some regex-replacement operations in your favourite editor. It shouldn't be too difficult. Is there a specific reason for preferring 0.5 over the 1.0 version?

**Comment by powerman** [comment]
uMatrix is able to import RequestPolicy 0.5 backup, but not 1.0. And yes, I work around this by converting rules using perl oneliner script.

**Comment by myrdd** [comment]
> uMatrix is able to import RequestPolicy 0.5 backup, but not 1.0.

True that RP 1.0 cannot import 0.5 backup files, but it can import 0.5
rules from the corresponding preferences. RP 1.0 automatically imports
the 0.5 rules on the first upgrade, or manually on
"about:requestpolicy?oldrules".

**Comment by myrdd** [comment]
copied from #892: TODO:

- [ ] provide functions in the settings
- [ ] provide functions on the "setup" page
